feature/src/main/java/com/terning/feature/filtering/filteringthree/FilteringThreeRoute.kt

```diff
@@ -36,50 +38,50 @@ fun FilteringThreeRoute(
     workingPeriod: Int,
     navigateUp: () -> Unit,
     navigateToStartHome: () -> Unit,
-    viewModel: FilteringViewModel = hiltViewModel(),
+    viewModel: FilteringThreeViewModel = hiltViewModel(),
 ) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+
     val context = LocalContext.current
     val lifecycleOwner = LocalLifecycleOwner.current
 
-    val currentYear = Calendar.getInstance().get(Calendar.YEAR)
-    val currentMonth = Calendar.getInstance().get(Calendar.MONTH)
-
-    var chosenYear by remember { mutableIntStateOf(currentYear) }
-    var chosenMonth by remember { mutableIntStateOf(currentMonth) }
-
     LaunchedEffect(key1 = true) {
-        viewModel.fetchGrade(grade = grade)
-        viewModel.fetchWorkingPeriod(workingPeriod = workingPeriod)
+        with(viewModel) {
+            updateGrade(grade = grade)
+            updateWorkingPeriod(workingPeriod = workingPeriod)
+        }
     }
 
-    LaunchedEffect(key1 = chosenYear, key2 = chosenMonth) {
-        viewModel.fetchStartYear(chosenYear)
-        viewModel.fetchStartMonth(chosenMonth)
+    LaunchedEffect(key1 = state.startYear, key2 = state.startMonth) {
+        with(viewModel) {
+            updateStartYear(state.startYear)
+            updateStartMonth(state.startMonth)
+        }
     }
 
     LaunchedEffect(viewModel.sideEffects, lifecycleOwner) {
         viewModel.sideEffects.flowWithLifecycle(lifecycle = lifecycleOwner.lifecycle)
             .collect { sideEffect ->
                 when (sideEffect) {
-                    is FilteringSideEffect.NavigateToStartHome -> navigateToStartHome()
-                    is FilteringSideEffect.ShowToast -> context.toast(sideEffect.message)
+                    is FilteringThreeSideEffect.NavigateToStartHome -> navigateToStartHome()
+                    is FilteringThreeSideEffect.ShowToast -> context.toast(sideEffect.message)
+                    is FilteringThreeSideEffect.NavigateUp -> navigateUp()
                 }
             }
     }
 
     FilteringThreeScreen(
-        navigateUp = { navigateUp() },
-        chosenYear = chosenYear,
-        chosenMonth = chosenMonth,
-        onNextClick = { viewModel.postFilteringWithServer() },
-        onYearChosen = { chosenYear = it },
-        onMonthChosen = { chosenMonth = it }
+        navigateUp = viewModel::navigateUp,
+        chosenYear = Calendar.getInstance().currentYear,
+        chosenMonth = Calendar.getInstance().currentMonth,
+        onNextClick = viewModel::postFilteringWithServer,
+        onYearChosen = { viewModel.updateStartYear(it) },
+        onMonthChosen = { viewModel.updateStartMonth(it) }
```

Comment:
```suggestion
        onYearChosen = viewModel::updateStartYear,
        onMonthChosen = viewModel::updateStartMonth
```
호출하는 함수가 하나이고, 인자의 순서가 맞다면 얘네도 "::"로 호출이 가능합니닷